### PR TITLE
Added defensive programming to willDestroy method

### PR DESCRIPTION
**Pull request by willjthorp**

### CODE REVIEW

File: addon/components/copy-button.js [hunk 1]
```diff
@@ -52,6 +52,10 @@ export default Component.extend({
   },
 
   willDestroyElement() {
-    get(this, 'clipboard').destroy();
+    const clipboard = get(this, 'clipboard').destroy();
+
+    if (clipboard) {
+      clipboard.destory();
```
Comment (netes) [inline addon/components/copy-button.js]:
```suggestion
      clipboard.destroy();
```

Comment (DarrenLehane) [inline addon/components/copy-button.js]:
```suggestion
      clipboard.destroy();
```

File: addon/components/copy-button.js [hunk 2]
```diff
@@ -52,6 +52,10 @@ export default Component.extend({
   },
 
   willDestroyElement() {
-    get(this, 'clipboard').destroy();
+    const clipboard = get(this, 'clipboard').destroy();
```
Comment (smnkrt) [inline addon/components/copy-button.js]: it seems `destroy` should not be called here 🤔

File: addon/components/copy-button.js (hunk 1 above)
Comment (willjthorp) [reply addon/components/copy-button.js]: 🤦‍♂️

File: addon/components/copy-button.js (hunk 2 above)
Comment (willjthorp) [reply addon/components/copy-button.js]: Yep my bad, I got a mix up with commits in the wrong cloned repo 🤦‍♂️updated now

**Pull request merged**